src/features/security.py

```diff
@@ -28,30 +32,18 @@ class Security(MetadataBase):
         "strict-transport-security": lambda s: "includesubdomains" in s.lower(),
     }
 
-    async def _start(self, website_data: WebsiteData) -> list[str]:
-        """
-        Run the header checks against the headers of given
-        site and return the field names of the checks that passed.
-        """
-        return [
-            field
-            for field, check in self.header_checks.items()
-            if field in website_data.headers and check(website_data.headers[field])
-        ]
+    async def setup(self):
+        pass
 
-    def _decide(self, website_data: WebsiteData) -> tuple[StarCase, Explanation]:
-        # fixme: Currently copy pasted from _start.
+    async def extract(self, site: WebsiteData, executor: Executor) -> tuple[StarCase, Explanation, set[str]]:
+        # No need to dispatch to the executor as this is a rather trivial computation
         passed = [
-            field
-            for field, check in self.header_checks.items()
-            if field in website_data.headers and check(website_data.headers[field])
+            field for field, check in self.header_checks.items() if field in site.headers and check(site.headers[field])
         ]
+        failed = set(self.header_checks.keys()) - set(passed)
```

Comment:
From the name of the variable I first though, this is a boolean. But it is not, it is a list of headers , which failed the security check, right? If so, please rename.

Reply:
done